packages/sanity/src/core/form/studio/assetSource/file/FileListView.tsx

```diff
@@ -1,56 +1,58 @@
 import React from 'react'
 import {Box, Card, Flex, Grid, Label, Spinner, Stack, useMediaIndex} from '@sanity/ui'
-import {Asset as AssetType} from '@sanity/types'
+import {Asset} from '@sanity/types'
+import {useTranslation} from '../../../../i18n'
 import {AssetRow} from './AssetRow'
 
 interface Props {
-  onClick?: (...args: any[]) => any
-  onKeyPress?: (...args: any[]) => any
-  onDeleteFinished: (...args: any[]) => any
-  assets: AssetType[]
+  onClick?: (event: React.MouseEvent) => void
+  onKeyPress?: (event: React.KeyboardEvent) => void
+  onDeleteFinished: (assetId: string) => void
+  assets: Asset[]
   isLoading?: boolean
-  selectedAssets: AssetType[]
+  selectedAssets: Asset[]
 }
 
 const STYLES_FILENAME = {paddingLeft: '2.2rem'}
 const STYLES_GRID = {gridTemplateColumns: '3fr 1fr 1fr 2fr 30px'}
 
-export function TableList(props: Props) {
+export function FileListView(props: Props) {
```

Comment:
seems risky to do renames of React components in a string extraction PR. Maybe fine.

Reply:
I tend to agree, but it was quite hard to follow the structure in in this code, so I did a refactor/cleanup as part of understanding it better. I've extracted this as a separate commit (b8739691218303d9ab9d2422b0e1c7e69ad373ac) that I'll fast track into next and doesn't need to be reviewed as part of this PR